cluster/replication/consumer.go

```diff
@@ -576,9 +591,17 @@ func (c *CopyOpConsumer) processFinalizingOp(ctx context.Context, op ShardReplic
 		return api.ShardReplicationState(""), ctx.Err()
 	}
 
-	if _, err := c.leaderClient.AddReplicaToShard(ctx, op.Op.TargetShard.CollectionId, op.Op.TargetShard.ShardId, op.Op.TargetShard.NodeId); err != nil {
-		logger.WithError(err).Error("failure while adding replica to shard")
-		return api.ShardReplicationState(""), err
+	if !replicaExists {
+		if _, err := c.leaderClient.AddReplicaToShard(ctx, op.Op.TargetShard.CollectionId, op.Op.TargetShard.ShardId, op.Op.TargetShard.NodeId); err != nil {
+			if strings.Contains(err.Error(), sharding.ErrReplicaAlreadyExists.Error()) {
+				// The replica already exists, this is not an error and it got updated after our sanity check
+				// due to eventual consistency of the sharding state.
+				logger.Debug("replica already exists, skipping")
+			} else {
+				logger.WithError(err).Error("failure while adding replica to shard")
+				return api.ShardReplicationState(""), err
+			}
+		}
```

Comment:
wdyt about merging the boolean logic here to:
```go
_, err := c.leaderClient.AddReplicaToShard(ctx, op.Op.TargetShard.CollectionId, op.Op.TargetShard.ShardId, op.Op.TargetShard.NodeId); err != nil && !replicaExists { ... }
```
to remove one instance of indentation?